Updated custom multi AddStairs/RemoveItem

JulianUO opened the pull request:
Modified the loop to search through iStairID in AddStairs (though using a short, no idea if that's the correct naming convention), now it is similar to what you can see in RemoveStairs. The last argument seems to be related to AddMulti but couldn't find a reason to keep it.

RemoveItem was modified to not remove the entire floor no matter what, client doesn't seem to like that, the server might remove the floor but you can commit and see the whole invalid items placed on it or whatever was removed from view by the server. I removed that bunch of code, maybe it would be the best to put it between some #ifdef or leave it commented.

Anyway, these are the differences I found between the 56d and X versions.
Tested on latest client (7.0.69?). Everything seems to work fine.

cbnolok commented:
The floor removing thing was added to fix issue #202, further info about the necessity of that behavior is in issue #323 discussion

JulianUO commented:
I couldn't reproduce any of the bugs in #202 in this commit.

I just had this issue but couldn't reproduce it: 
1. Added walls of base floor. 
2. Filled first floor with stone pavers. 
3. Went to base floor and added stairs.
4. Went to first floor and couldn't see the stone pavers.
Everything restored after commiting.
I'll retest later to see if can reproduce it.

JulianUO commented:
I've tried and tried but could not reproduce any bug.
Not even the #202 bugs.

Before this pull request, when I removed an invalid tile in 2nd/3rd floor, the entire floor was removed from me, but when committed, the invalid tiles were there.

cbnolok commented:
Did you do those tests with an updated version? Because i'm running latest and by removing an invalid floor tile (actually sphere's selection is wider, because it considers invalid every floor tile which has a stair below, i couldn't do better without revolutionizing the code) in 3rd floor it removes the full current floor

JulianUO commented:
This is recording with the latest build (just downloaded from the automated builds)
https://streamable.com/m2pmn

cbnolok commented:
So do you confirm it's working with this pull request?

JulianUO commented:
Did some testing and everything worked fine with this pull request.
Anyway, I fear the same, I'll add another commit with the code commented before you can pull it.
